src/components/List/index.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import isEqual from 'lodash.isequal';
+
+import styles from '@/components/List/List.css';
+
+import ListItem from '@/components/ListItem';
+
+type ListItemCategory = 'simple' | 'icon' | 'checkbox';
+type Size = 'large' | 'medium';
+
+const largeSize = 'large';
+
+interface Props<T> {
+  getItemIsSelected?: (item?: T) => boolean;
+  getItemLabel: (item: T) => React.ReactNode;
+  getItemIcon?: (item?: T) => React.ReactNode;
+  getItemId?: (item?: T) => React.ReactNode;
+  getItemName?: (item?: T) => string;
+  getItemValue: (item: T) => string | number | string[];
+  onClickItem?: (item: T) => void;
+  id: string;
+  label?: string;
+  listItemCategory: ListItemCategory;
+  options: [T];
+  selected?: T;
+  size: Size;
+  variablesClassName?: string;
+}
+
+const List = <T extends {}>(props: Props<T>) => {
+  const {
+    label,
+    size,
+    getItemIsSelected,
+    getItemLabel,
+    getItemIcon,
+    getItemName,
+    id,
+    onClickItem,
+    options,
+    selected,
+    getItemValue,
+    variablesClassName,
+    listItemCategory
+  } = props;
+
+  const [selectedItem, setSelectedItem] = useState(selected || null);
+
+  const sizeClass = `list--${size}`;
+  const listItemSizeClass = `list-item--${size}`;
+  const checkboxCategory = 'checkbox';
+
+  const checkBoxClick = item => {
+    if (isEqual(item, selectedItem)) {
+      setSelectedItem(null);
+    } else {
+      options.forEach(option => {
```

Comment:
you could use `find` instead of the `forEach` here
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find